contracts/SadaivBackup.sol

```diff
@@ -0,0 +1,65 @@
+//SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+contract SadaivBackup {
+  mapping(address => bool) public owners;
+
+  constructor() {
+    owners[msg.sender] = true;
+  }
+
+  modifier onlyOwner() {
+    require(owners[msg.sender], "Not authorized!");
+    _;
+  }
+
+  function makeOwner(address _address) public onlyOwner {
+    owners[_address] = true;
+  }
+
+  function checkIfOwner(address _address) public view returns (bool) {
+    return owners[_address];
+  }
+
+  struct Build {
+    string branch;
+    string commitMessage;
+    // string commitHash;
+    uint256 contributorGithubId;
+    string backupCid;
+    string changesCid; //future scope
+  }
+
+  struct Repository {
+    // uint256 id;
+    string name;
+    string fullname;
+    string description;
+    uint256 ownerGithubId;
+    uint256 size;
+    string defaultBranch;
+    string[] topics;
+    string language;
+  }
+
+  mapping(uint256 => Repository) public userRepos; //repoId => repo
+  mapping(uint256 => mapping(string => Build)) public repoBuilds; //repoId => commitHash => build

```

Comment:
RepoId -> commitHash -> build 

is good but multiple branches can commit with same hash. 

Repo

Commit Hash: 3x323
Branch A

now I use git pick this commit to new branch B and push
that will become 

Commit Hash: 3x323
Branch B

Reply:
so what do you suggest can be the identifier for the commit?

Reply:
I think it should be simply: repoid -> branch -> commitHash -> build 

I know this is quite nested. 

or maybe something like you hash branch + commitHash and use it as id.